common/determined_common/_logging.py

```diff
@@ -1,16 +1,207 @@
-    formatter = logging.Formatter("%(levelname)s: %(message)s")
-    handler.setFormatter(formatter)
-    root.addHandler(handler)
+    @property
+    def harness(self) -> logging.Logger:
+        return self._harness
+
+    @property
+    def resources(self) -> logging.Logger:
+        return self._resources
+
+
+# Loggers are normally global in python, and it would be a bit unexpected to behave differently.
+log = _Loggers()
+
+
+class _ProfilingThread(threading.Thread):
+    """
+    Call a profiling function periodically on a separate thread.
+    """
+
+    def __init__(self, period: float, profile_fn: Callable) -> None:
+        self.period = period
+        self.profile_fn = profile_fn
+        self.quitting = False
+        self.cond = threading.Condition()
+        super().__init__()
+
+    def run(self) -> None:
+        with self.cond:
+            while True:
+                self.cond.wait(timeout=self.period)
+                if self.quitting:
+                    break
+                self.profile_fn()
+
+    def quit(self) -> None:
+        self.quitting = True
+        with self.cond:
+            self.cond.notify_all()
+
+
+def read_level(level: Optional[str], default: int, limit: int = 0) -> int:
+    return max(logging._checkLevel(level if level is not None else default), limit)  # type: ignore
+
+
+class DebugConfig:
+    def __init__(
+        self,
+        root_log_level: Optional[str] = None,
+        storage_log_level: Optional[str] = None,
+        harness_log_level: Optional[str] = None,
+        debug_all_workers: bool = False,
+        horovod_verbose: bool = False,
+        nccl_debug: Optional[str] = None,
+        nccl_debug_subsys: Optional[str] = None,
+        resource_profile_period_sec: float = 0.0,
+        stack_trace_period_sec: float = 0.0,
+    ):
+        self._root_log_level = root_log_level
+        self._debug_all_workers = debug_all_workers
+        self._horovod_verbose = horovod_verbose
+        self._storage_log_level = storage_log_level
+        self._harness_log_level = harness_log_level
+        self._nccl_debug = nccl_debug
+        self._nccl_debug_subsys = nccl_debug_subsys
+        self._resource_profile_period_sec = resource_profile_period_sec
+        self._stack_trace_period_sec = stack_trace_period_sec
+
+        if 0.0 < self._resource_profile_period_sec < 0.1:
+            logging.warning(
+                "resource profiling with psutil and a resource_profile_period_sec less than .1 "
+                "seconds is unlikely to yield accurate results, adjusting to .1 seconds."
+            )
+            self._resource_profile_period_sec = 0.1
+
+        # The current workload, tracked here so we can emit it as part of the profiling output.
+        self._workload = ""
+
+    @classmethod
+    def from_config(cls, config: Union[None, bool, Dict[str, Any]]) -> "DebugConfig":
+        if config is None or config is False:
+            return cls()
+
+        if config is True:
```

Comment:
I didn't write it to be exactly equivalent.  My goal is more for `debug: True` to be generally useful to average users rather than identical behavior as before.  AFAIK nobody depends on the particular set of debug outputs from `debug: True`... mostly I see our customers pass us `debug: True` trial logs with messages like

> I ran with the debug: true flag on so there is a lot of debug spam

i.e. I think we have trained our users to turn `debug: true` on and pass it to us, but I don't think that they find it useful themselves.